pkg/lso/localvolumeset.go

```diff
@@ -0,0 +1,222 @@
+	// before the localVolumeSet object is created
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// NewLocalVolumeSetBuilder creates new instance of LocalVolumeSetBuilder.
+func NewLocalVolumeSetBuilder(apiClient *clients.Settings, name, nsname string) *LocalVolumeSetBuilder {
+	glog.V(100).Infof("Initializing new %s localVolumeSet structure in %s namespace", name, nsname)
+
+	builder := LocalVolumeSetBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeSet is empty")
+
+		builder.errorMsg = "localVolumeSet 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the localVolumeSet is empty")
+
+		builder.errorMsg = "localVolumeSet 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// PullLocalVolumeSet retrieves an existing localVolumeSet object from the cluster.
+func PullLocalVolumeSet(apiClient *clients.Settings, name, nsname string) (*LocalVolumeSetBuilder, error) {
+	glog.V(100).Infof(
+		"Pulling localVolumeSet object name:%s in namespace: %s", name, nsname)
+
+	builder := LocalVolumeSetBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeSet is empty")
+
+		builder.errorMsg = "localVolumeSet 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the localVolumeSet is empty")
+
+		builder.errorMsg = "localVolumeSet 'nsname' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("localVolumeSet oject %s doesn't exist in namespace %s", name, nsname)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches existing localVolumeDiscovery from cluster.
+func (builder *LocalVolumeSetBuilder) Get() (*LocalVolumeSetBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Pulling existing localVolumeSet with name %s under namespace %s from cluster",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	lvs := &lsoV1alpha1.LocalVolumeSet{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, lvs)
+
+	if err != nil {
+		return nil, err
+	}
+
+	builder.Object = lvs
+
+	return builder, nil
+}
+
+// Exists checks whether the given localVolumeSet exists.
+func (builder *LocalVolumeSetBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if localVolumeSet %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	_, err := builder.Get()
```

Comment:
Please replace it by:

builder.Object, err = builder.Get()

Reply:
same

Reply:
Even if it's going to be unused let's follow standard object structure.
Please update to:
builder.Object, err = builder.Get()